src/Shop.js

```diff
@@ -0,0 +1,84 @@
+class Shop {
+  constructor(items=[]){
+    this.items = items;
+  }
+
+  updateQuality() {
```

Comment:
These if/else statements with different functions are a good start. Let's try creating different classes that adhere to the Open Closed Principle and refactor this.

Reply:
For the creation of different classes, I discussed with Andrew and he suggested to make child classes of Item for the various products and put the update methods in each respective class. This is in line with the Open-Closed principle. How does that sound?

Reply:
Sure, it's kind of a really object oriented way of doing things now that I think about it and they only recently introduced classes actually. I guess the other option is to do it with currying functions but that might be a pretty complicated thing to explore for now.

Let's see what you come up with!